card/lib/card/query/abstract_query/tie.rb

```diff
@@ -0,0 +1,78 @@
+class Card
+  module Query
+    class AbstractQuery
+      # The "Tie" methods support tying two queries (CardQuery, ReferenceQuery, etc)
+      # together.  The "fasten" variable determines which tying strategy is used.
+      #
+      # We currently support three values for "fasten": :join, :exist, and :in
+      #
+      # In concept, here's how the different strategies would tie table A to table B
+      # in SQL assuming A.id = B.a_id
+      #
+      # - :join  ...  FROM A JOIN B ON A.id = B.a_id
+      # - :exist ...  FROM A WHERE EXISTS (SELECT * FROM B WHERE A.id = B.a_id ...)
+      # - :in    ...  FROM A WHERE A.id IN (SELECT B.a_id FROM B WHERE ...)
+      #
+      # The different strategies will return the same values but the relative speed is
+      # context dependent.
```

Comment:
This explains the tying strategies.

Reply:
Where are the tests that prove that it really works like that?

Reply:
coming.

Reply:
(they are there now)